apps/web/app/(app)/environments/[environmentId]/settings/api-keys/actions.ts

```diff
@@ -1,11 +1,31 @@
 "use server";
 
-import { deleteApiKey, createApiKey } from "@formbricks/lib/services/apiKey";
+import { authOptions } from "@/app/api/auth/[...nextauth]/authOptions";
+import { hasUserEnvironmentAccessCached } from "@formbricks/lib/environment/auth";
+import { deleteApiKey, createApiKey } from "@formbricks/lib/apiKey/service";
+import { canUserAccessApiKey } from "@formbricks/lib/apiKey/auth";
 import { TApiKeyCreateInput } from "@formbricks/types/v1/apiKeys";
+import { getServerSession } from "next-auth";
 
 export async function deleteApiKeyAction(id: string) {
-  return await deleteApiKey(id);
+  const session = await getServerSession(authOptions);
+  if (!session) throw new Error("You are not authorized to perform this action.");
```

Comment:
Please throw a `AuthorizationError` from `@formbricks/types/v1/errors`
Same in other places where you throw an error.